### PR TITLE
Refactoring Person to Item 

### PR DESCRIPTION
**Pull request by yixiann**
This PR was originally meant to create the list item function. However, it was impossible to do so with the current addressbook storing people instead of items.

Thus Person was replaced with Item. 

However, upon doing so, test cases failed across the application. 

A major refactoring was done. 

Upon refactoring, the commands present originally in AB3 was functional

This includes 
- Add
- Find
- Clear
- Delete
- Edit

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-  {:toc}
+- Table of Contents {:toc}
 
 ---
 
 ## **Acknowledgements**
 
-- {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+- {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+  original source as well}
```
Comment (bryanljx) [inline docs/DeveloperGuide.md:12]: The auto formatting messing the docs up again

File: docs/tutorials/AddRemark.md [hunk 2]
```diff
@@ -1,34 +1,37 @@
...
-title: "Tutorial: Adding a command"
+layout: page title: "Tutorial: Adding a command"
 ---
 
-Let's walk you through the implementation of a new command — `remark`.
+Let's walk you through the implementation of a new command — `remark`.
 
-This command allows users of the AddressBook application to add optional remarks to people in their address book and edit it if required. The command should have the following format:
+This command allows users of the AddressBook application to add optional remarks to people in their address book and
+edit it if required. The command should have the following format:
```
Comment (bryanljx) [inline docs/tutorials/AddRemark.md:8]: Again, the auto formatting messing things up. Not impt here tho cuz this is the tutorial/add-remark part.

File: docs/tutorials/TracingCode.md [hunk 3]
```diff
@@ -1,95 +1,124 @@
...
+> — Robert C. Martin Clean Code: A Handbook of Agile Software Craftsmanship
 
-When trying to understand an unfamiliar code base, one common strategy used is to trace some representative execution path through the code base. One easy way to trace an execution path is to use a debugger to step through the code. In this tutorial, you will be using the IntelliJ IDEA’s debugger to trace the execution path of a specific user command.
+When trying to understand an unfamiliar code base, one common strategy used is to trace some representative execution
+path through the code base. One easy way to trace an execution path is to use a debugger to step through the code. In
+this tutorial, you will be using the IntelliJ IDEA’s debugger to trace the execution path of a specific user command.
 
-* Table of Contents
-{:toc}
+* Table of Contents {:toc}
```
Comment (bryanljx) [inline docs/tutorials/TracingCode.md:13]: Ditto

File: src/main/java/seedu/foodrem/commons/core/Config.java [hunk 4]
```diff
@@ -54,10 +54,7 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        StringBuilder sb = new StringBuilder();
-        sb.append("Current log level : " + logLevel);
-        sb.append("\nPreference file Location : " + userPrefsFilePath);
-        return sb.toString();
+        return "Current log level : " + logLevel + "\nPreference file Location : " + userPrefsFilePath;
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/commons/core/Config.java:57]: Just curious, is there a reason for this change?

File: src/main/java/seedu/foodrem/commons/core/GuiSettings.java [hunk 5]
```diff
@@ -70,10 +70,7 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        StringBuilder sb = new StringBuilder();
-        sb.append("Width : " + windowWidth + "\n");
-        sb.append("Height : " + windowHeight + "\n");
-        sb.append("Position : " + windowCoordinates);
-        return sb.toString();
+        return "Width : " + windowWidth + "\n" + "Height : "
+                + windowHeight + "\n" + "Position : " + windowCoordinates;
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/commons/core/GuiSettings.java:74]: Ditto

File: src/main/java/seedu/foodrem/commons/core/index/Index.java [hunk 6]
```diff
@@ -45,10 +37,18 @@ public static Index fromOneBased(int oneBasedIndex) {
...
+        return zeroBasedIndex + 1;
+    }
+
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
-                || (other instanceof Index // instanceof handles nulls
-                && zeroBasedIndex == ((Index) other).zeroBasedIndex); // state check
+            || (other instanceof Index // instanceof handles nulls
+            && zeroBasedIndex == ((Index) other).zeroBasedIndex); // state check
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/commons/core/index/Index.java:52]: The auto formatting settings seem off. The style guide for cs2103t specifies 8 spaces for line breaks, not 4.

File: src/main/java/seedu/foodrem/commons/core/index/Index.java [hunk 7]
```diff
@@ -1,15 +1,15 @@
...
 
 /**
  * Represents a zero-based or one-based index.
- *
+ * <p>
  * {@code Index} should be used right from the start (when parsing in a new user input), so that if the current
  * component wants to communicate with another component, it can send an {@code Index} to avoid having to know what
  * base the other component is using for its index. However, after receiving the {@code Index}, that component can
  * convert it back to an int if the index will not be passed to a different component again.
  */
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/commons/core/index/Index.java:10]: Need verify this but is the <p> needed for cs2103t style guide for javadocs?

File: src/main/java/seedu/foodrem/commons/util/JsonUtil.java [hunk 8]
```diff
@@ -51,7 +51,8 @@ static <T> T deserializeObjectFromJsonFile(Path jsonFile, Class<T> classOfObject
     /**
      * Returns the Json object from the given file or {@code Optional.empty()} object if the file is not found.
      * If any values are missing from the file, default values will be used, as long as the file is a valid json file.
-     * @param filePath cannot be null.
+     *
+     * @param filePath                   cannot be null.
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/commons/util/JsonUtil.java:55]: Auto formatting again

File: src/main/java/seedu/foodrem/commons/util/JsonUtil.java [hunk 9]
```diff
@@ -102,8 +105,9 @@ public static <T> T fromJsonString(String json, Class<T> instanceClass) throws I
 
     /**
      * Converts a given instance of a class into its JSON data string representation
+     *
      * @param instance The T object to be converted into the JSON string
-     * @param <T> The generic type to create an instance of
+     * @param <T>      The generic type to create an instance of
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/commons/util/JsonUtil.java:110]: Auto formatting again

File: src/main/java/seedu/foodrem/commons/util/StringUtil.java [hunk 10]
```diff
@@ -14,14 +14,15 @@ public class StringUtil {
...
+     * Ignores case, but a full word match is required.
+     * <br>examples:<pre>
      *       containsWordIgnoreCase("ABc def", "abc") == true
      *       containsWordIgnoreCase("ABc def", "DEF") == true
      *       containsWordIgnoreCase("ABc def", "AB") == false //not a full word match
      *       </pre>
+     *
      * @param sentence cannot be null
-     * @param word cannot be null, cannot be empty, must be a single word
+     * @param word     cannot be null, cannot be empty, must be a single word
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/commons/util/StringUtil.java:25]: Auto formatting again

File: src/main/java/seedu/foodrem/commons/util/StringUtil.java [hunk 11]
```diff
@@ -31,11 +32,10 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
         checkArgument(!preppedWord.isEmpty(), "Word parameter cannot be empty");
         checkArgument(preppedWord.split("\\s+").length == 1, "Word parameter should be a single word");
 
-        String preppedSentence = sentence;
-        String[] wordsInPreppedSentence = preppedSentence.split("\\s+");
+        String[] wordsInPreppedSentence = sentence.split("\\s+");
 
         return Arrays.stream(wordsInPreppedSentence)
-                .anyMatch(preppedWord::equalsIgnoreCase);
+            .anyMatch(preppedWord::equalsIgnoreCase);
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/commons/util/StringUtil.java:38]: Ditto

File: src/main/java/seedu/foodrem/model/ModelManager.java [hunk 12]
```diff
@@ -1,56 +1,56 @@
...
 
 /**
- * Represents the in-memory model of the address book data.
+ * Represents the in-memory model of FoodRem data.
  */
 public class ModelManager implements Model {
     private static final Logger logger = LogsCenter.getLogger(ModelManager.class);
 
-    private final AddressBook addressBook;
+    private final FoodRem addressBook;
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/model/ModelManager.java:22]: Please rename the variable name addressBook to foodRem as well

File: src/main/java/seedu/foodrem/model/UserPrefs.java [hunk 13]
```diff
@@ -78,10 +79,7 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        StringBuilder sb = new StringBuilder();
-        sb.append("Gui Settings : " + guiSettings);
-        sb.append("\nLocal data file location : " + addressBookFilePath);
-        return sb.toString();
+        return "Gui Settings : " + guiSettings + "\nLocal data file location : " + addressBookFilePath;
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/model/UserPrefs.java:82]: Ditto

File: src/main/java/seedu/foodrem/model/item/ItemBoughtDate.java [hunk 14]
```diff
@@ -35,8 +38,8 @@ public ItemBoughtDate(String dateString) {
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
-                || (other instanceof ItemBoughtDate // instanceof handles nulls
-                && itemDate.equals(((ItemBoughtDate) other).itemDate)); // state check
+            || (other instanceof ItemBoughtDate // instanceof handles nulls
+            && itemDate.equals(((ItemBoughtDate) other).itemDate)); // state check
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/model/item/ItemBoughtDate.java:42]: Ditto

File: src/main/java/seedu/foodrem/model/item/ItemBoughtDate.java [hunk 15]
```diff
@@ -52,6 +55,6 @@ public int hashCode() {
      */
     @Override
     public String toString() {
-        return itemDate.format(DateTimeFormatter.ofPattern(DATE_OUTPUT_PATTERN_REGEX));
+        return itemDate.toString();
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/model/item/ItemBoughtDate.java:58]: Is `DATE_OUTPUT_PATTERN_REGEX` not used anymore?

File: src/main/java/seedu/foodrem/model/item/ItemExpiryDate.java [hunk 16]
```diff
@@ -35,8 +38,8 @@ public ItemExpiryDate(String dateString) {
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
-                || (other instanceof ItemExpiryDate // instanceof handles nulls
-                && itemDate.equals(((ItemExpiryDate) other).itemDate)); // state check
+            || (other instanceof ItemExpiryDate // instanceof handles nulls
+            && itemDate.equals(((ItemExpiryDate) other).itemDate)); // state check
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/model/item/ItemExpiryDate.java:42]: Ditto

File: src/main/java/seedu/foodrem/model/item/ItemExpiryDate.java [hunk 17]
```diff
@@ -52,6 +55,6 @@ public int hashCode() {
      */
     @Override
     public String toString() {
-        return itemDate.format(DateTimeFormatter.ofPattern(DATE_OUTPUT_PATTERN_REGEX));
+        return itemDate.toString();
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/model/item/ItemExpiryDate.java:58]: Ditto

File: src/main/java/seedu/foodrem/model/item/itemvalidator/ItemBoughtDateValidator.java [hunk 18]
```diff
@@ -15,28 +13,29 @@ public class ItemBoughtDateValidator {
...
     // Validation for year
     private static final int MIN_YEAR = 1000;
     private static final String MESSAGE_FOR_YEAR_TOO_SMALL =
-            String.format("Year should be larger than %d.", MIN_YEAR);
+        String.format("Year should be larger than %d.", MIN_YEAR);
 
     private static final int MAX_YEAR = 2100;
     private static final String MESSAGE_FOR_YEAR_TOO_LARGE =
-            String.format("Year should be less than %d.", MAX_YEAR);
+        String.format("Year should be less than %d.", MAX_YEAR);
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/model/item/itemvalidator/ItemBoughtDateValidator.java:26]: Ditto

File: src/main/java/seedu/foodrem/model/util/SampleDataUtil.java [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+            new Item(new ItemName("Potatoes"),
+                new ItemQuantity("10"),
+                new ItemUnit("kg"),
+                new ItemBoughtDate("2022-11-11"),
+                new ItemExpiryDate("2022-11-11")),
+            new Item(new ItemName("Cucumbers"),
+                new ItemQuantity("2000"),
+                new ItemUnit("grams"),
+                new ItemBoughtDate("2022-11-11"),
+                new ItemExpiryDate("2022-11-11"))
```
Comment (bryanljx) [inline src/main/java/seedu/foodrem/model/util/SampleDataUtil.java:32]: Should create sample data with different expiry date from bought date. Possibly add some items that do not have expiry dates as well.

**Review by bryanljx — CHANGES_REQUESTED** [review]
Generally looks good. However, the auto formatting is messing up a lot of files and since AddressBook is changed to FoodRem, corresponding var names in these files should also be renamed to foodRem. 

Besides that, should be fine. I will continue reviewing after you fix these first. Left the changes to test cases to go.

**Comment by yixiann** [comment]
Duplicated : https://github.com/AY2223S1-CS2103T-W16-2/tp/pull/151

**Pull request closed without merging**